src/log.cpp

```diff
@@ -18,16 +18,22 @@
 #include "log.h"
 #include "profiler.h"
 
-
-const char* const Log::LEVEL_NAME[] = {NULL, "TRACE", "DEBUG", "INFO", "WARN", "ERROR"};
+const char* const Log::LEVEL_NAME[] = {
+#define LOG_LEVEL_STR(level) #level,
+LOG_LEVELS(LOG_LEVEL_STR)
+#undef LOG_LEVEL_STR
+    "none",
+    NULL
+};
```

Comment:
I'm not a fan of higher order macros (though they are often used, for example, in HotSpot code).
They definitely make the code harder to read and they are not well supported in IDEs.
I can jusify such macros when they significantly reduce the amount of boilerplate, but this is not the case here.

Reply:
I've removed the higher order macro.